### PR TITLE
XYZ-62: Character count enforced for previous message when moving to a new channel

### PR DESCRIPTION
**Pull request by MusikPolice**
#### Summary
Forced a re-check of the character count whenever the text box receives focus. This event fires when the user clicks the mouse to focus the textbox, but also when the user switches channels (in the centre column), and when the user clicks the reply arrow to enter a new thread in the RHS. 

#### Ticket Link
https://mattermost.atlassian.net/browse/XYZ-62

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

### CODE REVIEW

File: tests/components/textbox.test.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
```
Comment (MusikPolice) [inline tests/components/textbox.test.jsx]: I was hoping to make these tests more comprehensive, but I had difficulty getting them to properly simulate user actions, and I was coming up against the release deadline. The behaviour that this PR fixes is perhaps best tested with Selenium.

File: utils/utils.jsx [hunk 2]
```diff
@@ -1554,3 +1554,14 @@ export function copyToClipboard(e, data) {
     document.execCommand('copy');
     document.body.removeChild(textArea);
 }
+
+export function generateRandomString(length) {
```
Comment (jespino) [inline utils/utils.jsx]: I think this function would be in tests/helpers/, but I'm 2/5 about it.

Comment (MusikPolice) [reply utils/utils.jsx]: Ah, I didn't know about `tests/helpers/`. I can move it into `client-test-helper.jsx`

Comment (MusikPolice) [reply utils/utils.jsx]: Done in https://github.com/mattermost/mattermost-webapp/pull/715/commits/cdae7298f0c73e19ec2e515fddf976cd8253e9de

**Pull request merged**